lib/table_kitty.ex

```diff
@@ -11,28 +11,228 @@
 # limitations under the License.
 defmodule TableKitty do
   @moduledoc """
-  Documentation for `TableKitty`.
-  """
+  `TableKitty` prints text-based tables based on tabular data.
 
-  @doc """
-  Print the table if the same is valid.
+  Tabular data means that we expect something that is a row-based
+  or a column-based table. In fact we expect the data to implement
+  the `Table.Reader` protocol. This is already implemented for
+  lists and maps. See the [table package](https://hex.pm/packages/table)
+  on Hex.pm to discover the libs that implement this protocol.
+
+  The main function is `build/2` that is going to return a result
+  with an IO data if successful. IO data has the advantage over
+  strings because it will postpone the construction of the final
+  string. You can use `IO.iodata_to_binary/1` to get a string out of it.
+
+  For a simple print, there are `print/2` and `TableKitty.inspect/2`.
+  The first one will print and return `:ok` or will return `{:error, exception}`.
+  And the second will print and return the given tabular data or
+  will raise an error.
+
+  ## Examples
+
+      iex> data = [city: ["Tokyo", "Delhi", "Shanghai", "São Paulo", "Mexico City"], pop: [37, 28, 25, 21, 21]]
+      iex> io_data = TableKitty.build!(data, title: "Largest cities", headers: %{pop: "Population in millions"})
+      iex> IO.iodata_to_binary(io_data)
+      \"\"\"
+      +--------------------------------------+
+      |            Largest cities            |
+      +-------------+------------------------+
+      | city        | Population in millions |
+      +=============+========================+
+      | Tokyo       | 37                     |
+      +-------------+------------------------+
+      | Delhi       | 28                     |
+      +-------------+------------------------+
+      | Shanghai    | 25                     |
+      +-------------+------------------------+
+      | São Paulo   | 21                     |
+      +-------------+------------------------+
+      | Mexico City | 21                     |
+      +-------------+------------------------+
+      \"\"\"
+
+  This one is a "lighter" version of the above:
+
+      iex> data = [city: ["Tokyo", "Delhi", "Shanghai", "São Paulo", "Mexico City"], pop: [37, 28, 25, 21, 21]]
+      iex> io_data = TableKitty.build!(data, display_horizontal_divisor: false)
+      iex> IO.iodata_to_binary(io_data)
+      \"\"\"
+      +-------------+-----+
+      | city        | pop |
+      +=============+=====+
+      | Tokyo       | 37  |
+      | Delhi       | 28  |
+      | Shanghai    | 25  |
+      | São Paulo   | 21  |
+      | Mexico City | 21  |
+      +-------------+-----+
+      \"\"\"
```

Comment:
Seeing now, this version without the horizontal lines is much cleaner. I will change the default to it.